### PR TITLE
fix: Add initial in-statements indent support (Updated)

### LINKED ISSUE #7884: statement_indentation is indenting already correct line
### Description

When running `statement_indentation` on a multiline if, if the first line is missing indentation, the following lines are getting useless extra indentation. This is especially a blocking point for https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7813 since
```php
if (true
    || false
) {
    // foo
}
```
is fixed to
```php
if (
true
    || false
) {
    // foo
}
```
and then to
```php
if (
    true
        || false // This line is indented for no reason
) {
    // foo
}
```

When I expect
```php
if (
    true
    || false // This line is untouched
) {
    // foo
}
```

### Runtime version

`3.51.0`

### Configuration file

`'statement_indentation' => true`

### Code snippet that reproduces the problem

  ```php
  <?php
if (
true
    || false
) {
    // foo
}
  ```

### PR DESCRIPTION
**Pull request by VincentLanglet**
Close #7901
Close #7884

Hi @Wirone @keradus, I'm really interested in the PR https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901 since it's needed for https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7813 ;

I rebased the PR and fix the tests.

### CODE REVIEW

**Review by kubawerlos — APPROVED** [review]
The changes look technically good and I like them, but (😁):

I am a bit afraid that this change might make a fuss when released, so I'm leaving the final decision about merging to @keradus.

File: src/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixer.php
Comment (SpacePossum) [inline src/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixer.php:67]:
```
            'Enforce API encapsulation in an inheritance architecture. '
                .'If you want to override a method, use the Template method pattern.',
            'Risky when overriding `public` methods of `abstract` classes.'
```
this added indent on the middle line is intended?

Comment (kubawerlos) [reply src/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixer.php:67]: It is, it adds intent before the dot. The line `'Risky when overriding...` is another argument.

Comment (SpacePossum) [reply src/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixer.php:67]: Thanks for the details. I'm not 100% sure I follow the rational for the change the rule makes all the time, but that should not be of concern by itself per se.
I do wonder, as the rule is part of PER-CS2; do we find the rule with the change as proposed here in the PER-CS2 standard, or should this new behavior be opt-in through configuration?

Comment (VincentLanglet) [reply src/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixer.php:67]: If you're interested in the history @SpacePossum

I was working on a FullyMultilineFixer for PER-CS2 and got an issue https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7813/files#r1531647667

```
if ($a
    || $b) {
    // foo
}
```
was changed to
```
if (
$a
    || $b) {
    // foo
}
```
but then the StatementIndentationFixer change the code to
```
if (
    $a
        || $b) {
    // foo
}
```
this is explained here https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7884#issue-2185751932

So I understand correctly the implementation made by the original author in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901, it requires that the StatementIndentationFixer does not rely anymore on the existing indentation, but have an (opinionated ?) rule for indentation and inner-indentation.

But maybe indeed, some config could be added for the inner-indentation. (It can also be done in another PR, the iteration seems doable).

**Review by Wirone — APPROVED** [review]
Just like in [#7901](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901#pullrequestreview-2133612975) and like @kubawerlos - I like these changes, and I want them to be incorporated, but the last word goes to @keradus 🤷.

**Comment by VincentLanglet** [comment]
Friendly ping @keradus ; it would be helpful for https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7813 to move this PR forward.

**Comment by VincentLanglet** [comment]
> Just like in [#7901](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901#pullrequestreview-2133612975) and like @kubawerlos - I like these changes, and I want them to be incorporated, but the last word goes to @keradus 🤷.

Friendly ping @keradus ; I already got @Wirone @kubawerlos approval.
This PR is a blocker for another one ; what is missing to move this one forward ?

Thanks for your time.

**Comment by keradus** [comment]
Is the change good?
- mostly

Is the change working properly?
- not fully, you can find some cases that got wrong indent [here](https://github.com/symfony/symfony/pull/58925#discussion_r1848192142) (look for other comments like that)

Is the change supposed to be enabled for all, or behind feature configuration?
- looking on things like [this](https://github.com/symfony/symfony/pull/58925/files#r1848191359), it may have the cases when ppl won't be happy to have this style change enforced and feature config needed. Happy to see reacton of Sf community on that PR

**Comment by VincentLanglet** [comment]
Since I dunno how this fixer work, I'll need to rely on @julienfalque time (you were waiting for feedback on https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901#issuecomment-2182766843).

I was just helping to move this forward but this seems to big for me...

**Comment by keradus** [comment]
@Wirone I originally re-opened Julien's PR https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7901#issuecomment-2486932031

keep any of them , but probably no need for both :D

**Pull request closed without merging**
